src/experimental-bots/filler/fillerMultithreaded.ts

```diff
@@ -320,79 +336,121 @@ export class FillerMultithreaded {
-						);
-						break;
-					case 'triggerableNodes':
-						if (this.dryRun) {
-							logger.info(`Triggerable node received`);
-						} else {
-							this.triggerNodes(msg.data);
-						}
-						this.lastTryFillTimeGauge?.setLatestValue(
-							Date.now(),
-							metricAttrFromUserAccount(
-								user.getUserAccountPublicKey(),
-								user.getUserAccount()
-							)
-						);
-						break;
-					case 'fillableNodes':
-						if (this.dryRun) {
-							logger.info(`Fillable node received`);
-						} else {
-							this.fillNodes(msg.data);
+
+		for (const marketIndexes of this.marketIndexes) {
+			logger.info(
+				`${this.name}: Spawning dlobBuilder for marketIndexes: ${marketIndexes}`
+			);
+			const dlobBuilderArgs = [
+				`--market-type=${this.config.marketType}`,
+				`--market-indexes=${marketIndexes.map(String)}`,
+			];
+			const dlobBuilderProcess = spawnChildWithRetry(
+				'./src/experimental-bots/filler/dlobBuilder.ts',
+				dlobBuilderArgs,
+				'dlobBuilder',
+				(msg: any) => {
+					switch (msg.type) {
+						case 'initialized':
+							for (const dlobBuilder of this.dlobBuilders) {
+								if (
+									dlobBuilder.marketIndexes.every(
+										(val, index) => val === msg.data[index]
+									)
+								) {
+									dlobBuilder.ready = true;
+									logger.info(
+										`${logPrefix} dlobBuilderProcess initialized and acknowledged`
+									);
+									break;
+								}
+							}
+							break;
+						case 'triggerableNodes':
+							if (this.dryRun) {
+								logger.info(`Triggerable node received`);
+							} else {
+								this.triggerNodes(msg.data);
+							}
+							this.lastTryFillTimeGauge?.setLatestValue(
+								Date.now(),
+								metricAttrFromUserAccount(
+									user.getUserAccountPublicKey(),
+									user.getUserAccount()
+								)
+							);
+							break;
+						case 'fillableNodes':
+							if (this.dryRun) {
+								logger.info(`Fillable node received`);
+							} else {
+								this.fillNodes(msg.data);
+							}
+							this.lastTryFillTimeGauge?.setLatestValue(
+								Date.now(),
+								metricAttrFromUserAccount(
+									user.getUserAccountPublicKey(),
+									user.getUserAccount()
+								)
+							);
+							break;
+						case 'health':
+							this.dlobHealthy = msg.data.healthy;
+							break;
+					}
+				},
+				'[FillerMultithreaded]'
+			);
+
+			this.dlobBuilders.push({
+				process: dlobBuilderProcess,
+				ready: false,
+				marketIndexes: marketIndexes.map(Number),
+			});
+
+			logger.info(
+				`dlobBuilder spawned with pid: ${dlobBuilderProcess.pid} marketIndexes: ${dlobBuilderArgs}`
+			);
+		}
+
+		const routeMessageToDlobBuilder = (msg: any) => {
+			for (const dlobBuilder of this.dlobBuilders) {
+				if (dlobBuilder.marketIndexes.includes(msg.data.marketIndex)) {
+					if (typeof dlobBuilder.process.send == 'function') {
+						if (dlobBuilder.ready) {
+							dlobBuilder.process.send(msg);
+							return;
 						}
```

Comment:
small nit but we can probably keep a map of market index -> dlob process to avoid this for loop. we should def not have multiple markets exist on diff dlob builder processes in any case. might be good to make that check explicit too to avoid bad configs